### PR TITLE
[prometheus-pushgateway] Add support for containerSecurityContext to fix #1741

### LINKED ISSUE #1741: [prometheus-pushgateway] Can not specify securityContext at container level
### Describe the bug a clear and concise description of what the bug is.

As per https://github.com/prometheus-community/helm-charts/blob/prometheus-pushgateway-1.10.1/charts/prometheus-pushgateway/values.yaml#L120 , securityContext is applied at pod spec (as in https://github.com/prometheus-community/helm-charts/blob/prometheus-pushgateway-1.10.1/charts/prometheus-pushgateway/templates/deployment.yaml#L82).
But users should be able to specify securityContext at container level too, as both spec support different level of settings. 

Moreover, manually adding some of the settings (below) to the deployment confirms they indeed work fine,

`
$ kubectl get pod prometheus-pushgateway-5d77df776-qqhh8 -n monitoring -o=jsonpath='{.spec.containers[].securityContext}'
{"allowPrivilegeEscalation":false,"readOnlyRootFilesystem":true,"runAsNonRoot":true,"runAsUser":65534}
`

### What's your helm version?

version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"clean", GoVersion:"go1.14.9"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"windows/amd64"} Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.7", GitCommit:"1f86634ff08f37e54e8bfcd86bc90b61c98f84d4", GitTreeState:"clean", BuildDate:"2021-11-18T00:11:23Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

prometheus-pushgateway

### What's the chart version?

v1.10.1

### What happened?

securityContext settings get applied to pod spec not and not the container spec.

### What you expected to happen?

Ideally we should be able to define 2 securityContext settings that will operate at container and pod spec level,

securityContext:
   <container level parameters>
podSecurityContext: 
    <pod level parameters>

### How to reproduce it?

_No response_

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

helm install prometheus-pushgateway prometheus-community/prometheus-pushgateway --version v1.10.1 --values values.yaml

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by AshutoshNirkhe**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Please change minor chart version, not app version.

**Comment by AshutoshNirkhe** [comment]
> Please change minor chart version, not app version.

Thanks @monotek I have updated it now, could you please check again ?

File: charts/prometheus-pushgateway/values.yaml [hunk 1]
```diff
@@ -126,6 +126,14 @@ securityContext:
   runAsUser: 65534
   runAsNonRoot: true
 
+## Security context to be added to push-gateway containers
+## Having a separate variable as securityContext differs for pods and containers.
+containerSecurityContext:
+  allowPrivilegeEscalation: false
```
Comment (monotek) [inline charts/prometheus-pushgateway/values.yaml:132]: imho these vlaues should not be added by default or updating the chart for some users might break.

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Comment by djsly** [comment]
thanks a lot @AshutoshNirkhe  for this work

**Pull request merged**